File: 프로그래머스/lv2/42860. 조이스틱/조이스틱.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(string name) {
    int answer = 0;
    int len=name.length();
    int target_idx;
    int move=len-1;    
    for(int i=0;i<name.size();i++)
    {
        answer+=min(name[i]-'A','Z'-name[i]+1); // 현재 i위치 상하를 이용하여 조절한 하여 마춘 후 
        target_idx=i+1; //다음 위치
        
        while(target_idx<len&&name[target_idx]=='A') // 다음 변경할 것을 찾기위한 이동 
        {
            target_idx++; //연속으로 A 넘어감
        }
        
        //여기서 target_idx는 다음 고쳐야할 위치
        int a = i; //왼쪽에서 
        int b = len-target_idx; //오른쪽 끝에서 타겟까지 거리
        // 0 -> a -> a -> b 
        // 0 -> b -> b -> a , 
        //둘 중 최솟 값
        move=min(move,a+b+min(a,b)); //a+b는
    }
    
    return answer+move;
}

File: 프로그래머스/lv2/42860. 조이스틱/조이스틱.py
def solution(name):
    l = len(name)
    now = "A" * l
    zOrd = ord("Z")
    aOrd = ord("A")
    idx = 0
    answer = 0
    lr = l -1
    
    for i in range(l):
        target = ord(name[i])
        answer += min(target-aOrd,zOrd-target+1)
        idx = i+1
        while (idx < l and name[idx] == "A"):
            idx += 1
        
        left = i
        right = l - idx
        
        print(left,right)
        
        lr =  min(lr,left+right + min(left,right))
        
        
        
        
    

    return answer +lr